lib/video_info/providers/vkontakte.rb

```diff
@@ -52,11 +58,15 @@ def duration
       private
 
       def _set_data_from_api
-        uri = open(url, options)
+        url = URI.parse("https://vk.com/al_video.php")
+        options["act"] = "show"
+        options["al"] = "1"
+        options["video"] = "#{@video_owner}_#{@video_id}"
+        resp = Net::HTTP.post_form(url, options)
         if RUBY_VERSION.to_i < 2
-          Iconv.iconv('utf-8', 'cp1251', uri.read)[0]
+          Iconv.iconv("UTF-8//TRANSLIT//IGNORE", "cp1251", resp.body)[0]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.